### ISSUE TITLE
`cargo fmt`: `cargo manifest` failed

### ISSUE THREAD

**Issue by nhynes** [rust-lang-nursery/rustfmt#2379]
platform: Ubuntu 16.04
`rustup --version`: `rustup 1.9.0 (57fc3c087 2018-01-04)`
`cargo +nightly --version`: `cargo 0.26.0-nightly (6a8eb71f6 2018-01-13)`
`cargo +nightly fmt --version`: `0.3.6-nightly ( )`
`rustfmt --version`: `0.3.6-nightly ( )`

Steps to reproduce:

```
rustup update nightly
cargo +nightly install rustfmt-nightly --version 0.3.6 --force
cargo +nightly fmt
```

Observe:
```
`cargo manifest` failed.
usage: cargo fmt [options]
...
```
I tried an internet search for "cargo manifest failed" but nothing relevant is returned. Where have I gone wrong?

`rustfmt **/*.rs` works, however.

**Comment by nhynes** [comment]
Looking at the code suggested running `cargo metadata` which said that a path dependency was missing. It might be nice to have a better error message, but hopefully this helps anyone else who runs into the issue!

**Comment by joshlf** [comment]
I recently opened https://github.com/rust-lang/cargo/issues/5043 about this issue (and I was redirected to this repository). I think that there should be an open issue to improve the error message here; would it be kosher to migrate that issue to this repository, or would you rather track in this issue?

**Comment by nhynes** [comment]
@joshlf I'm sorry that you had a similar issue! The reason the [error message](https://github.com/rust-lang-nursery/rustfmt/blob/8f8ee9e66040df6b02bb7cd6bb777ab2fc4069d8/cargo-fmt/src/main.rs#L370) is unhelpful is because the error returned by [cargo_metadata](https://github.com/oli-obk/cargo_metadata/blob/master/src/lib.rs#L250) is unhelpful. The best solution would probably to have `cargo_metadata::metadata` capture stdout as the `err.description`.

**Comment by joshlf** [comment]
OK gotcha. Would you like me to open an issue about that? Unfortunately I don't have time to work on a full PR right now.

**Comment by pronebird** [comment]
Do you guys have to update rustfmt for this change to kick in? I just recently stumbled upon the same and the error message is the same -- unhelpful

**Comment by nhynes** [comment]
The fix was posted o `cargo_metadata` v 0.5.0. `rustfmt` master currently uses v0.5.1, so it should be good to go.
